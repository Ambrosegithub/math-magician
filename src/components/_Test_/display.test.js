// import { render, screen, userEvent } from '@testing-library/react';
// import Calculator from '../calculator';

// describe('Test rendering and user interaction', () => {
//   it('renders correctly', () => {
//     const tree = render(<Calculator />);
//     expect(tree).toMatchSnapshot();
//   });

//   it('User interaction', () => {
//     render(<Calculator />);

//     userEvent.click(screen.getByRole('button', { name: '2' }));
//     userEvent.click(screen.getByRole('button', { name: '1' }));
//     userEvent.click(screen.getByRole('button', { name: '+' }));
//     userEvent.click(screen.getByRole('button', { name: '7' }));
//     userEvent.click(screen.getByRole('button', { name: '=' }));
//     expect(screen.handleClickEvent()).toHaveBeenCalledTimes(5);
//   });
// });
import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import displayBtn from '../mock/display';

test('click', () => {
  render(
    <displayBtn />,
  );

  userEvent.click(screen.getByText('Check'));
  expect(screen.getByLabelText('Check')).toBeChecked();
});
