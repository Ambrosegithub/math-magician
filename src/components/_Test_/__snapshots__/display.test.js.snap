// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test display Butoons Test Function Button component 1`] = `
<displayBtn
  e={[MockFunction]}
  name="Ambrose"
/>
`;

exports[`Test rendering and user interaction renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="container"
      >
        <h1
          class="calc-title"
        >
          Let's do some mathematics!
        </h1>
        <div
          class="calc-box"
        >
          <div
            class="screen"
          >
            <span
              class="output"
            >
              0
            </span>
          </div>
          <div
            class="buttons"
          >
            <button
              class="calc"
              type="button"
            >
              AC
            </button>
            <button
              class="calc"
              type="button"
            >
              +/-
            </button>
            <button
              class="calc"
              type="button"
            >
               
              %
            </button>
            <button
              class="calc active"
              type="button"
            >
              <span>
                รท
              </span>
            </button>
            <button
              class="calc"
              type="button"
            >
              7
            </button>
            <button
              class="calc"
              type="button"
            >
              8
            </button>
            <button
              class="calc"
              type="button"
            >
              9
            </button>
            <button
              class="calc active"
              type="button"
            >
              x
            </button>
            <button
              class="calc"
              type="button"
            >
              4
            </button>
            <button
              class="calc"
              type="button"
            >
              5
            </button>
            <button
              class="calc"
              type="button"
            >
              6
            </button>
            <button
              class="calc active"
              type="button"
            >
              -
            </button>
            <button
              class="calc"
              type="button"
            >
              1
            </button>
            <button
              class="calc"
              type="button"
            >
              2
            </button>
            <button
              class="calc"
              type="button"
            >
              3
            </button>
            <button
              class="calc active"
              type="button"
            >
              +
            </button>
            <button
              class="zero"
              type="button"
            >
              0
            </button>
            <button
              class="calc"
              type="button"
            >
              .
            </button>
            <button
              class="calc active"
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="container"
    >
      <h1
        class="calc-title"
      >
        Let's do some mathematics!
      </h1>
      <div
        class="calc-box"
      >
        <div
          class="screen"
        >
          <span
            class="output"
          >
            0
          </span>
        </div>
        <div
          class="buttons"
        >
          <button
            class="calc"
            type="button"
          >
            AC
          </button>
          <button
            class="calc"
            type="button"
          >
            +/-
          </button>
          <button
            class="calc"
            type="button"
          >
             
            %
          </button>
          <button
            class="calc active"
            type="button"
          >
            <span>
              รท
            </span>
          </button>
          <button
            class="calc"
            type="button"
          >
            7
          </button>
          <button
            class="calc"
            type="button"
          >
            8
          </button>
          <button
            class="calc"
            type="button"
          >
            9
          </button>
          <button
            class="calc active"
            type="button"
          >
            x
          </button>
          <button
            class="calc"
            type="button"
          >
            4
          </button>
          <button
            class="calc"
            type="button"
          >
            5
          </button>
          <button
            class="calc"
            type="button"
          >
            6
          </button>
          <button
            class="calc active"
            type="button"
          >
            -
          </button>
          <button
            class="calc"
            type="button"
          >
            1
          </button>
          <button
            class="calc"
            type="button"
          >
            2
          </button>
          <button
            class="calc"
            type="button"
          >
            3
          </button>
          <button
            class="calc active"
            type="button"
          >
            +
          </button>
          <button
            class="zero"
            type="button"
          >
            0
          </button>
          <button
            class="calc"
            type="button"
          >
            .
          </button>
          <button
            class="calc active"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
